### PR TITLE
fix(component version): loading versions

### PR DESCRIPTION
**Pull request by luvkapur**
## Proposed Changes

- Only fetch what is needed on initial load
- Refactor useComponent hooks - extract fetching logs to separate hook
- Update `VersionsRelatedDropdowns` and `VersionDropdown` with a more robust clean API

### CODE REVIEW

File: scopes/component/component/ui/menu/menu.tsx [hunk 1]
```diff
@@ -122,91 +130,213 @@ export function ComponentMenu({
...
+        skip: loadingFromProps || skip,
+        customUseComponent: useComponent,
+      };
+      const {
+        component,
+        loading: loadingComponent,
+        componentLogs = {},
+      } = useComponentQuery(host, componentId, fetchOptions);
+      const logs = componentLogs?.logs;
+      const loading = React.useMemo(
```
Comment (JoshK2) [inline scopes/component/component/ui/menu/menu.tsx:199]: Remove "React".

File: scopes/component/component/ui/menu/menu.tsx [hunk 2]
```diff
@@ -122,91 +130,213 @@ export function ComponentMenu({
...
+export const defaultLoadCurrentVersion: (props: VersionRelatedDropdownsProps) => UseComponentVersion = (props) => {
+  return (_props) => {
+    const { skip, version: _version } = _props || {};
+    const { snaps, tags, currentVersion, loading } = props.useComponent?.({ skip, id: props.componentId }) ?? {};
+    const version = _version ?? currentVersion;
+    const isTag = React.useMemo(() => semver.valid(version), [loading, version]);
+    if (isTag) {
+      return React.useMemo(() => tags?.find((tag) => tag.tag === version), [loading, tags?.length, version]);
+    }
+    return React.useMemo(() => snaps?.find((snap) => snap.version === version), [loading, snaps?.length, version]);
```
Comment (JoshK2) [inline scopes/component/component/ui/menu/menu.tsx:244]: Remove "React"

File: scopes/component/component/ui/menu/menu.tsx [hunk 3]
```diff
@@ -122,91 +130,213 @@ export function ComponentMenu({
...
-  const lanes = lanesModel?.getLanesByComponentId(component.id)?.filter((lane) => !lane.id.isDefault()) || [];
+  const isNew = tags?.length === 0 && snaps?.length === 0;
+
   const localVersion = isWorkspace && !isNew && (!viewedLane || lanesModel?.isViewingCurrentLane());
 
   const currentVersion =
-    isWorkspace && !isNew && !location?.search.includes('version') ? 'workspace' : component.version;
+    isWorkspace && !isNew && !location?.search.includes('version') ? 'workspace' : _currentVersion ?? '';
+
+  const consumeMethodProps = React.useMemo(() => {
```
Comment (JoshK2) [inline scopes/component/component/ui/menu/menu.tsx:286]: Remove "React"

File: scopes/component/ui/version-dropdown/version-dropdown-placeholder.module.scss [hunk 4]
```diff
@@ -4,94 +4,109 @@
...
+    text-overflow: ellipsis;
+  }
+
   .versionTimestamp {
-    min-width: fit-content;
-    margin-right: 2px;
-    text-align: right;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
```
Comment (JoshK2) [inline scopes/component/ui/version-dropdown/version-dropdown-placeholder.module.scss:98]: Consider using https://bit.cloud/teambit/design/ui/styles/ellipsis
This component have both Component and Class only exported.

File: scopes/component/ui/version-dropdown/version-dropdown.tsx [hunk 5]
```diff
@@ -7,162 +7,291 @@ import { UserAvatar } from '@teambit/design.ui.avatar';
...
-    if (currentLane?.components.some((c) => c.version === currentVersion))
-      return tabs.findIndex((tab) => tab.name === 'LANE');
-    if ((snaps || []).some((snap) => snap.version === currentVersion))
-      return tabs.findIndex((tab) => tab.name === 'SNAP');
-    return 0;
-  };
+  const { snaps, tags, loading: loadingVersions } = useVersions?.() || {};
+  const loading = loadingFromProps || loadingVersions;
+
+  const tabs = React.useMemo(
```
Comment (JoshK2) [inline scopes/component/ui/version-dropdown/version-dropdown.tsx:216]: Remove "React"

File: scopes/component/ui/version-dropdown/version-info/version-info.module.scss [hunk 6]
```diff
@@ -4,41 +4,57 @@
...
+  line-height: 24px;
 }
+
 .versionTimestamp {
   min-width: fit-content;
   margin-right: 2px;
   text-align: right;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
```
Comment (JoshK2) [inline scopes/component/ui/version-dropdown/version-info/version-info.module.scss:16]: Same here, consider using: https://bit.cloud/teambit/design/ui/styles/ellipsis

File: scopes/component/ui/version-dropdown/version-info/version-info.module.scss [hunk 7]
```diff
@@ -4,41 +4,57 @@
...
-  &.tag {
-    min-width: fit-content;
-  }
-  &.snap {
-    min-width: 20%;
-  }
+  flex-grow: 1;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
```
Comment (JoshK2) [inline scopes/component/ui/version-dropdown/version-info/version-info.module.scss:42]: Same here, consider using: https://bit.cloud/teambit/design/ui/styles/ellipsis

File: scopes/component/ui/version-dropdown/version-info/version-info.module.scss [hunk 8]
```diff
@@ -4,41 +4,57 @@
...
 }
 
-.commitMessage,
-.emptyMessage {
-  min-width: 50%;
+.commitMessage {
+  flex-grow: 1;
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
```
Comment (JoshK2) [inline scopes/component/ui/version-dropdown/version-info/version-info.module.scss:50]: Same here, consider using: https://bit.cloud/teambit/design/ui/styles/ellipsis
You can use only the class from this component if you prefer.

File: scopes/component/ui/version-dropdown/version-info/version-info.module.scss (hunk 7 above)
Comment (JoshK2) [reply scopes/component/ui/version-dropdown/version-info/version-info.module.scss:42]: I see that this class `versionName` is already used in the Ellipsis component, so you can just remove the unnecessary style.

File: scopes/component/ui/version-dropdown/version-info/version-info.tsx [hunk 9]
```diff
@@ -46,16 +55,20 @@ export function VersionInfo({
...
+    <div ref={ref || currentVersionRef}>
       <MenuLinkItem active={isCurrent} href={href} className={styles.versionRow}>
         <div className={styles.version}>
           <UserAvatar size={24} account={author} className={styles.versionUserAvatar} showTooltip={true} />
-          <Ellipsis className={classNames(styles.versionName, tag && styles.tag, !tag && styles.snap)}>
-            {version}
-          </Ellipsis>
-          {version === latestVersion && <VersionLabel className={styles.label} status="latest" />}
-          {showDetails && commitMessage(message)}
+          <Ellipsis className={classNames(styles.versionName)}>{formattedVersion}</Ellipsis>
```
Comment (JoshK2) [inline scopes/component/ui/version-dropdown/version-info/version-info.tsx:69]: You can remove the ellipsis style from the `versionName` class.

File: scopes/component/ui/version-dropdown/version-info/version-info.tsx [hunk 10]
```diff
@@ -46,16 +55,20 @@ export function VersionInfo({
...
+          {isLatest && <VersionLabel status="latest" />}
+          <CommitMessage message={message} showDetails={showDetails} />
         </div>
         <Ellipsis className={styles.versionTimestamp}>
```
Comment (JoshK2) [inline scopes/component/ui/version-dropdown/version-info/version-info.tsx:73]: Same here, you can remove the ellipsis style from the class, it’s already exist in the `Ellipsis` component.

File: scopes/ui-foundation/use-box/menu/install.tsx [hunk 11]
```diff
@@ -13,27 +13,42 @@ export type InstallProps = {
...
       {...rest}
       content={
         <>
-          <div>{`Install ${componentName} with ${packageManager}`}</div>
-          <TooltipCopybox content={copyString} />
+          <div className={!isInstallable && styles.disabled}>
+            {!isInstallable && <Icon className={styles.warnIcon} of="warn-circle" />}
+            {isInstallable && `Install ${componentName} with ${packageManager}`}
+            {!isInstallable && 'Installation unavailable: This component has not been built yet'}
+          </div>
```
Comment (JoshK2) [inline scopes/ui-foundation/use-box/menu/install.tsx:37]: Maybe change to:
```
!isInstallable ? (
<>
<Icon className={styles.warnIcon} of="warn-circle" />
Installation unavailable: This component has not been built yet
</>
) : Install ${componentName} with ${packageManager}
```

**Review by JoshK2 — DISMISSED** [review]

**Review by JoshK2 — DISMISSED** [review]

**Review by davidfirst — APPROVED** [review]

**Pull request merged**